backend/Dockerfile.prod

```diff
@@ -28,4 +28,4 @@ COPY --from=build  /home/gradle/graphql-api/backend/insights.jar insights.jar
 RUN addgroup -S appgroup && adduser -S appuser -G appgroup
 USER appuser
 
-ENTRYPOINT ["java", "-javaagent:insights.jar", "-jar", "app.jar"]
+ENTRYPOINT ["java", "-javaagent:insights.jar", "-XX:MaxRAMPercentage=80", "-jar", "app.jar"]
```

Comment:
How does this work with multiple containers / autoscaling? A larger heap solves our OOM issues related to Liquibase, since only one container can hold the lock at a time, but does this open the door to OOM issues when multiple containers are growing their heap simultaneously?

Reply:
I will be honest...I do not know for sure. The documentation on this from Azure's side is quite poor, and a lot of it is abstracted away from us. 

From what I have been able to gather, the App Service plan will intelligently distribute available memory across all apps...not individual instances...in the plan. Right now, our plan shows that we are running [2 apps](https://portal.azure.com/#@cdc.onmicrosoft.com/resource/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-simple-report-prod/providers/Microsoft.Web/serverfarms/prime-simple-report-appserviceplan-prod).

Each container should be able to tell how much memory is assigned to it, and adjust accordingly, based on this configuration. I'm seeing reports, though, that Azure [limits container memory to 250 MB per container](https://docs.microsoft.com/en-us/answers/questions/2721/azure-app-for-containers-custom-container-memory-a.html), though...which is frustrating. That also seems to align with what we see in [dev](https://portal.azure.com/#@cdc.onmicrosoft.com/resource/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-simple-report-dev/providers/Microsoft.Web/sites/simple-report-api-dev/metrics), where we have roughly 500 MB of an average working set across two containers.

I really hope this doesn't mean our containers are actually allocating a quarter of 250 MB for their heap sizes...

Dev seems like it's been slowly allocating more memory since I deployed this changeset earlier this morning, so this may have lifted a previous limit. I don't honestly know how high that usage percentage will go, though.

I think we may need to experiment with manually configuring the memory limits on these containers to go above 250MB. If that has a change, I can get you a better answer. Unfortunately, unless the docs suddenly update, or unless I missed something major, experimentation may be our only way through this. I've tried getting our containers to output their memory settings on dev, but the log entries were swallowed by Azure. I may be able to turn the log level up to get those numbers, so we can see effects and get more information.

Either that, or we can move to K8S. 🙃

Reply:
@rin-skylight what memory is the limit of 250 MB per container for? total memory or JVM heap memory?

Reply:
It looks like 250MB per container, which is almost certainly too low for the JVM, and 25% of that as a heap size is even worse. Here are some thoughts:
- Can we edit the PR title real quick? It defaulted to the branch name.
- I think you're correct in that the App Plan is shared with every resource in that plan, so each container is using that shared RAM pool.
- So in one sense, this is a distributed systems/Java question rather than Azure - what happens if we run into a scenario where each container grows the heap to the max size? I think it's probably dueling OOMs and garbage collection but I don't know.
- In light of that, can we also set a `MinRAMPercentage` to the same value? I think I'd rather be greedy and claim the max heap size instead of potentially dealing with GC pauses when growing/shrinking the heap.
- Did the 80% number come from any particular source? The JVM needs enough memory to operate outside of the heap, so I wonder if that number is a little high.
- If we're going to claim heap as a percentage, we should also limit the max memory of a container, both in Azure and locally.

Reply:
Let me tackle these in order:

- Title modified!
- Agreed.
- I can't say. Azure has minimal documentation on this...the App Service paradigm black-boxes a lot of this away from us. That's one of the unfortunate tradeoffs we make with it...Azure handles the orchestration and underlying management and maintenance, at the expense of control.
- We can certainly try that, yes! I will test and push a change that does that, just to make sure. Sources I have read show that the GC pause can be massively reduced by doing so, so great callout on that!
- The 80% number was me being a little aggressive, I just realized. I do have a source for this: https://dzone.com/articles/best-practices-java-memory-arguments-for-container#:~:text=Bit%20Server%20VM-,You%20can%20see%20when%20the%20docker%20container's%20memory%20is%20set,(approximately%20half%20of%201GB). This recommends ensuring that the container has 25% more RAM allocated than the heap size, to allow for that overhead. I will adjust that metric accordingly.
- Agreed. I'll work on getting that added in.

Reply:
Following up...`MinRAMPercentage` is ignored when containers have access to more than 200MB of memory. `InitialRAMPercentage` accomplishes the same ask, so it has been used, instead.

One issue with limiting memory usage...both in Azure, and locally. We are using Compose v3. From the research I have seen, memory limiting is only available in v3 if you use Docker Swarm. Leveraging Compose for local development, or in simple stacks, doesn't have a way to force memory or CPU limitation. This caused quite an uproar in the community when this took place. v2 still retains that, so we could downgrade if it were absolutely essential. Azure seems like it will intelligently manage the amount of RAM available to each of our App Service instances, but the documentation is nebulous, at best.

Reply:
Thanks for the answers! I think these changes are in the ballpark, so let's try it out.